### DISCUSSION TITLE
Print the variable value

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#19642] [Q&A Modules: General] [answered]
I am quite new to MOOSE and C++. So just wondering how can I  make sure the value given in input file is properly passed to the kernel file or material file.
I mean is there any method to print the value or matrix in the kernel file?

Thanks in advance.

**Comment by friedmud** [comment, marked as answer]
Just for your own purposes of feeling comfortable that MOOSE is doing the right thing you can always add a quick `std::cout << _my_param_value << std::endl;` in `computeQpResidual` to make sure the value is coming through.  Obviously, this is not something you want to have in there for a long time (it will cause millions of print outs!) - but it will give you the confidence that you're doing things correctly in the beginning :-)

You can also see what MOOSE is reading from your input file by using `--show-input` on the command-line.  This will cause MOOSE to echo out what it thinks it read from your input file (and show you all of the default values for all parameters you didn't specify).

**Reply by avtarsinghh1991** [reply]
Thanks.

when I am using 
>> --show-input

it says command not found.

**Reply by friedmud** [reply]
That is a command-line option for your executable... so do:

`./my_moose_program -i my_input  --show-input`

**Reply by avtarsinghh1991** [reply]
Thank you very much. It helps a lot.
